saleor/graphql/channel/__init__.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+from typing import Any, Optional
+
+from django.db.models import QuerySet
+
+
+@dataclass
+class ChannelContext:
+    node: Any
```

Comment:
This works, but consider:

```python
from typing import Generic, TypeVar

N = TypeVar("N")

@dataclass
class ChannelContext(Generic[N]):
    node: N
    channel_slug: Optional[str] = None
```

then you can use strong typing:

```python
def resolve_foo(root: ChannelContext[Product], ...):
    root.node # <- will be typed automatically

def resolve_bar(...) -> ChannelContext[ProductVariant]:
    return ...
```

A similar construct could be used for ChannelQsContext (assuming we have `django-stubs` installed for mypy).

Reply:
👍